src/isar/apis/models/start_mission_definition.py

```diff
@@ -113,6 +92,64 @@ def to_isar_mission(mission_definition: StartMissionDefinition) -> Mission:
     return isar_mission
 
 
+def check_for_duplicate_ids(items: Union[List[Task], List[STEPS]]):
+    duplicate_ids = get_duplicate_ids(items=items)
+    if len(duplicate_ids) > 0:
+        raise MissionPlannerError(
+            f"Failed to create as there were duplicate IDs which is not allowed "
+            f"({duplicate_ids})"
+        )
+
+
+def generate_steps(task) -> List[STEPS]:
+    try:
+        steps: List[STEPS] = []
+
```

Comment:
I would argue it is better to have this outside the `try catch`, to minimize the code inside it

Reply:
Done